{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Dangerous Mutations\n",
    "\n",
    "--------\n",
    "\n",
    "## Clinical Disease Data Parsing\n",
    "\n",
    "My manager comes to me Monday morning and says “I figured out our next step; we are going to pivot from an online craft store and become a data center for genetic disease information! I found **ClinVar** which is a repository that contains expert curated data, and it is free for the taking. This is a gold mine! Look at the file and tell me what gene and mutation combinations are classified as dangerous.”\n",
    "\n",
    "In this project, I am going to extract the dangerous mutations, the final deliverable is a dataframe table to summarize these mutations with following information:\n",
    "\n",
    "1) Gene name\n",
    "\n",
    "2) Mutation ID number\n",
    "\n",
    "3) Mutation Position (chromosome & position)\n",
    "\n",
    "4) Mutation value (reference & alternate bases)\n",
    "\n",
    "5) Clinical significance (CLNSIG)\n",
    "\n",
    "6) Disease that is implicated\n",
    "\n",
    "**Requirements**\n",
    "\n",
    "1) Limit the output to the first 100 harmful mutations and tell my manager how many total harmful mutations were found in the file\n",
    "\n",
    "2) Use the \"clinvar_final.txt\" data which could be downloaded from this link: https://drive.google.com/file/d/1Zps0YssoJbZHrn6iLte2RDLlgruhAX1s/view?usp=sharing \n",
    "\n",
    "3) Filter out junk and lines with no mutation data. Just focus on the data to deliver the result. \n",
    "\n",
    "\n",
    "Note:\n",
    "\n",
    "* The unit of observation of this dataset is one row per mutation."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "### VCF file description (Summarized from version 4.1)\n",
    "\n",
    "```\n",
    "* The VCF specification:\n",
    "\n",
    "VCF is a text file format which contains meta-information lines, a header line, and then data lines each containing information about a position in the genome. The format also can contain genotype information on samples for each position.\n",
    "\n",
    "* Fixed fields:\n",
    "\n",
    "There are 8 fixed fields per record. All data lines are **tab-delimited**. In all cases, missing values are specified with a dot (‘.’). \n",
    "\n",
    "1. CHROM - chromosome number\n",
    "2. POS - position DNA nuceleotide count (bases) along the chromosome\n",
    "3. ID - The unique identifier for each mutation\n",
    "4. REF - reference base(s)\n",
    "5. ALT - alternate base(s)\n",
    "6. FILTER - filter status\n",
    "7. QUAL - quality\n",
    "8. INFO - a semicolon-separated series of keys with values in the format: <key>=<data>\n",
    "\n",
    "```\n",
    "### Applicable INFO field specifications\n",
    "\n",
    "```\n",
    "GENEINFO = <Gene name>\n",
    "CLNSIG =  <Clinical significance>\n",
    "CLNDN = <Disease name>\n",
    "```\n",
    "\n",
    "### Sample ClinVar data (vcf file format - not exactly the same as the file to download!)\n",
    "\n",
    "```\n",
    "##fileformat=VCFv4.1\n",
    "##fileDate=2019-03-19\n",
    "##source=ClinVar\n",
    "##reference=GRCh38\t\t\t\t\t\t\t\n",
    "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n",
    "1\t949523\trs786201005\tC\tT\t.\t.\tGENEINFO=ISG15;CLNSIG=5\n",
    "1\t949696\trs672601345\tC\tCG\t.\t.\tGENEINFO=ISG15;CLNSIG=5;CLNDN=Cancer\n",
    "1\t949739\trs672601312\tG\tT\t.\t.\tGENEINFO=ISG15;CLNDBN=Cancer\n",
    "1\t955597\trs115173026\tG\tT\t.\t.\tGENEINFO=AGRN;CLNSIG=2; CLNDN=Cancer\n",
    "1\t955619\trs201073369\tG\tC\t.\t.\tGENEINFO=AGG;CLNDN=Heart_dis \n",
    "1\t957640\trs6657048\tC\tT\t.\t.\tGENEINFO=AGG;CLNSIG=3;CLNDN=Heart_dis \n",
    "1\t976059\trs544749044\tC\tT\t.\t.\tGENEINFO=AGG;CLNSIG=0;CLNDN=Heart_dis \n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#fileformat=VCFv4.1\n",
      "#fileDate=2019-03-01\n",
      "#source=ClinVar\n",
      "#reference=GRCh38\n",
      "#ID=<Description=\"ClinVar Variation ID\">\n",
      "#INFO=<ID=AF_ESP,Number=1,Type=Float,Description=\"allele frequencies from GO-ESP\">\n",
      "#INFO=<ID=AF_EXAC,Number=1,Type=Float,Description=\"allele frequencies from ExAC\">\n",
      "#INFO=<ID=AF_TGP,Number=1,Type=Float,Description=\"allele frequencies from TGP\">\n",
      "#INFO=<ID=ALLELEID,Number=1,Type=Integer,Description=\"the ClinVar Allele ID\">\n",
      "#INFO=<ID=CLNDN,Number=.,Type=String,Description=\"ClinVar's preferred disease name for the concept specified by disease identifiers in CLNDISDB\">\n",
      "#INFO=<ID=CLNDNINCL,Number=.,Type=String,Description=\"For included Variant : ClinVar's preferred disease name for the concept specified by disease identifiers in CLNDISDB\">\n",
      "#INFO=<ID=CLNDISDB,Number=.,Type=String,Description=\"Tag-value pairs of disease database name and identifier, e.g. OMIM:NNNNNN\">\n",
      "##INFO=<ID=CLNDISDBINCL,Number=.,Type=String,Description=\"For included Variant: Tag-value pairs of disease database name and identifier, e.g. OMIM:NNNNNN\">\n",
      "#INFO=<ID=CLNHGVS,Number=.,Type=String,Description=\"Top-level (primary assembly, alt, or patch) HGVS expression.\">\n",
      "#INFO=<ID=CLNREVSTAT,Number=.,Type=String,Description=\"ClinVar review status for the Variation ID\">\n",
      "#INFO=<ID=CLNSIG,Number=.,Type=String,Description=\"Clinical significance for this single variant\">\n",
      "#INFO=<ID=CLNSIGCONF,Number=.,Type=String,Description=\"Conflicting clinical significance for this single variant\">\n",
      "#INFO=<ID=CLNSIGINCL,Number=.,Type=String,Description=\"Clinical significance for a haplotype or genotype that includes this variant. Reported as pairs of VariationID:clinical significance.\">\n",
      "#INFO=<ID=CLNVC,Number=1,Type=String,Description=\"Variant type\">\n",
      "#INFO=<ID=CLNVCSO,Number=1,Type=String,Description=\"Sequence Ontology id for variant type\">\n",
      "#INFO=<ID=CLNVI,Number=.,Type=String,Description=\"the variant's clinical sources reported as tag-value pairs of database and variant identifier\">\n",
      "#INFO=<ID=DBVARID,Number=.,Type=String,Description=\"nsv accessions from dbVar for the variant\">\n",
      "#INFO=<ID=GENEINFO,Number=1,Type=String,Description=\"Gene(s) for the variant reported as gene symbol:gene id. The gene symbol and id are delimited by a colon (:) and each pair is delimited by a vertical bar (|)\">\n",
      "#INFO=<ID=MC,Number=.,Type=String,Description=\"comma separated list of molecular consequence in the form of Sequence Ontology ID|molecular_consequence\">\n",
      "#INFO=<ID=ORIGIN,Number=.,Type=String,Description=\"Allele origin. One or more of the following values may be added: 0 - unknown; 1 - germline; 2 - somatic; 4 - inherited; 8 - paternal; 16 - maternal; 32 - de-novo; 64 - biparental; 128 - uniparental; 256 - not-tested; 512 - tested-inconclusive; 1073741824 - other\">\n",
      "#INFO=<ID=RS,Number=.,Type=String,Description=\"dbSNP ID (i.e. rs number)\">\n",
      "#INFO=<ID=SSR,Number=1,Type=Integer,Description=\"Variant Suspect Reason Codes. One or more of the following values may be added: 0 - unspecified, 1 - Paralog, 2 - byEST, 4 - oldAlign, 8 - Para_EST, 16 - 1kg_failed, 1024 - other\">\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014O42</td>\n",
       "      <td>475283</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00546;AF_EXAC=0.00165;AF_TGP=0.00619;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1O14122</td>\n",
       "      <td>542074</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00015;AF_EXAC=0.00010;ALLELEID=514926...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014179</td>\n",
       "      <td>542075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=514896;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>475278</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00515;AF_EXAC=0.00831;AF_TGP=0.00339;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>1014228</td>\n",
       "      <td>402986</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.40158;AF_EXAC=0.37025;AF_TGP=0.33886;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>1014255</td>\n",
       "      <td>571208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=556509;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>1014276</td>\n",
       "      <td>568195</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=556512;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1014316</td>\n",
       "      <td>161455</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=171289;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>1014359</td>\n",
       "      <td>161454</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00001;ALLELEID=171288;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS      ID REF ALT FILTER QUAL  \\\n",
       "0      1  1014O42  475283   G   A      .    .   \n",
       "1      1  1O14122  542074   C   T      .    .   \n",
       "2      1  1014143  183381   C   T      .    .   \n",
       "3      1  1014179  542075   C   T      .    .   \n",
       "4      1  1014217  475278   C   T      .    .   \n",
       "5      1  1014228  402986   G   A      .    .   \n",
       "6      1  1014255  571208   G   A      .    .   \n",
       "7      1  1014276  568195   G   A      .    .   \n",
       "8      1  1014316  161455   C  CG      .    .   \n",
       "9      1  1014359  161454   G   T      .    .   \n",
       "\n",
       "                                                INFO  \n",
       "0  AF_ESP=0.00546;AF_EXAC=0.00165;AF_TGP=0.00619;...  \n",
       "1  AF_ESP=0.00015;AF_EXAC=0.00010;ALLELEID=514926...  \n",
       "2  ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "3  ALLELEID=514896;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "4  AF_ESP=0.00515;AF_EXAC=0.00831;AF_TGP=0.00339;...  \n",
       "5  AF_ESP=0.40158;AF_EXAC=0.37025;AF_TGP=0.33886;...  \n",
       "6  ALLELEID=556509;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "7  ALLELEID=556512;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "8  ALLELEID=171289;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "9  AF_EXAC=0.00001;ALLELEID=171288;CLNDISDB=MedGe...  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Load Raw Data\n",
    "import pandas as pd\n",
    "import io\n",
    "import os\n",
    "import re\n",
    "\n",
    "pd.set_option('display.max_rows', 1000)\n",
    "pd.options.display.float_format = '{:,.2f}'.format\n",
    "\n",
    "# Load data to DataFrame, exclude any row start with #\n",
    "with open('clinvar_final.txt', 'r') as fin:\n",
    "    lines = [line for line in fin if not line.startswith('#')]\n",
    " \n",
    "with open('clinvar_final.txt', 'r') as fin:\n",
    "    headers = [line for line in fin if line.startswith('#')]\n",
    "    \n",
    "\n",
    "df = pd.read_csv(io.StringIO(''.join(lines)),\n",
    "                 dtype={'CHROM': int, 'POS': str, 'ID': int, 'REF': str, 'ALT': str,\n",
    "                        'FILTER': str,'QUAL': str, 'INFO': str},\n",
    "                 sep='\\t')\n",
    "print(''.join(headers))\n",
    "df.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mutation_ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>mutation_position</th>\n",
       "      <th>mutation_value_ref</th>\n",
       "      <th>mutation_value_alt</th>\n",
       "      <th>clinical_significant</th>\n",
       "      <th>disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>475283</td>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>542074</td>\n",
       "      <td>1</td>\n",
       "      <td>1014122</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>183381</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>542075</td>\n",
       "      <td>1</td>\n",
       "      <td>1014179</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>475278</td>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102316</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>403908</td>\n",
       "      <td>3</td>\n",
       "      <td>179210507</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102317</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>526648</td>\n",
       "      <td>3</td>\n",
       "      <td>179210511</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102318</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>526640</td>\n",
       "      <td>3</td>\n",
       "      <td>179210515</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cowden_syndrome|Hereditary_cancer-predisposing...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102319</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>246681</td>\n",
       "      <td>3</td>\n",
       "      <td>179210516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102320</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>259958</td>\n",
       "      <td>3</td>\n",
       "      <td>179210538</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Cowden_syndrome|not_specified</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102321 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene_name  mutation_ID  chromosome  mutation_position  \\\n",
       "0        ISG15:9636       475283           1            1014042   \n",
       "1        ISG15:9636       542074           1            1014122   \n",
       "2        ISG15:9636       183381           1            1014143   \n",
       "3        ISG15:9636       542075           1            1014179   \n",
       "4        ISG15:9636       475278           1            1014217   \n",
       "...             ...          ...         ...                ...   \n",
       "102316  PIK3CA:5290       403908           3          179210507   \n",
       "102317  PIK3CA:5290       526648           3          179210511   \n",
       "102318  PIK3CA:5290       526640           3          179210515   \n",
       "102319  PIK3CA:5290       246681           3          179210516   \n",
       "102320  PIK3CA:5290       259958           3          179210538   \n",
       "\n",
       "       mutation_value_ref mutation_value_alt    clinical_significant  \\\n",
       "0                       G                  A                  Benign   \n",
       "1                       C                  T  Uncertain_significance   \n",
       "2                       C                  T              Pathogenic   \n",
       "3                       C                  T  Uncertain_significance   \n",
       "4                       C                  T                  Benign   \n",
       "...                   ...                ...                     ...   \n",
       "102316                  A                  G  Uncertain_significance   \n",
       "102317                  T                  C           Likely_benign   \n",
       "102318                  A                  C  Uncertain_significance   \n",
       "102319                  A                  G  Uncertain_significance   \n",
       "102320                  A                  T           Likely_benign   \n",
       "\n",
       "                                             disease_name  \n",
       "0       Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "1       Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "2       Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "3       Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "4       Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "...                                                   ...  \n",
       "102316                                    Cowden_syndrome  \n",
       "102317                                    Cowden_syndrome  \n",
       "102318  Cowden_syndrome|Hereditary_cancer-predisposing...  \n",
       "102319                                    Cowden_syndrome  \n",
       "102320                      Cowden_syndrome|not_specified  \n",
       "\n",
       "[102321 rows x 8 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Extract Gene name, clincal significant and disease name from raw data\n",
    "\n",
    "# the gene name is the value after the \"GENEINFO=\" and before the next \";\" in the \"INFO\"\n",
    "df['GENEINFO'] = df.INFO.str.extract(r'((?<=GENEINFO=).*)',\\\n",
    "                                      expand=False).str.replace(r'(;.*)','')\n",
    "\n",
    "# the clinical significant is the value after the \"CLNSIG=\" and before the next \";\" in the \"INFO\"\n",
    "df['CLNSIG'] = df.INFO.str.extract(r'((?<=CLNSIG=).*)',\\\n",
    "                                      expand=False).str.replace(r'(;.*)','')\n",
    "\n",
    "# the disease name is the value after the \"CLNDN=\" and before the next \";\" in the \"INFO\"\n",
    "df['CLNDN'] = df.INFO.str.extract(r'((?<=CLNDN=).*)',\\\n",
    "                                      expand=False).str.replace(r'(;.*)','')\n",
    "\n",
    "# Mutation position: replace \"O\" by '0' and change data type to 'int' \n",
    "df['POS'] = df.POS.str.replace(r'([Oo])','0')\n",
    "df = df.astype({'POS': 'int'})\n",
    "\n",
    "# Change order of columns & rename\n",
    "df = df[['GENEINFO','ID','CHROM','POS','REF','ALT','CLNSIG','CLNDN']]\n",
    "df.columns = ['gene_name', 'mutation_ID', 'chromosome','mutation_position',\n",
    "             'mutation_value_ref', 'mutation_value_alt','clinical_significant','disease_name']\n",
    "\n",
    "# Replace missing values with \"Not_Given\"\n",
    "df.fillna('Not_Given', inplace=True)\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Clinical significant values:\n",
      "\n",
      "count                     102321\n",
      "unique                        43\n",
      "top       Uncertain_significance\n",
      "freq                       47980\n",
      "Name: clinical_significant, dtype: object\n",
      "\n",
      "Uncertain_significance                                                                      47980\n",
      "Likely_benign                                                                               17885\n",
      "Pathogenic                                                                                  12313\n",
      "Likely_pathogenic                                                                            6269\n",
      "Benign                                                                                       6138\n",
      "Conflicting_interpretations_of_pathogenicity                                                 5404\n",
      "Benign/Likely_benign                                                                         3338\n",
      "Not_Given                                                                                    1797\n",
      "Pathogenic/Likely_pathogenic                                                                  854\n",
      "risk_factor                                                                                    98\n",
      "association                                                                                    70\n",
      "drug_response                                                                                  41\n",
      "Affects                                                                                        31\n",
      "Conflicting_interpretations_of_pathogenicity,_risk_factor                                      14\n",
      "other                                                                                          12\n",
      "Pathogenic,_risk_factor                                                                        11\n",
      "Pathogenic/Likely_pathogenic,_other                                                             8\n",
      "Conflicting_interpretations_of_pathogenicity,_other                                             8\n",
      "Benign/Likely_benign,_risk_factor                                                               7\n",
      "Pathogenic/Likely_pathogenic,_risk_factor                                                       6\n",
      "protective                                                                                      6\n",
      "Likely_pathogenic,_risk_factor                                                                  5\n",
      "protective,_risk_factor                                                                         4\n",
      "Pathogenic,_Affects                                                                             2\n",
      "Pathogenic,_other                                                                               2\n",
      "Benign/Likely_benign,_other                                                                     1\n",
      "Likely_pathogenic,_other                                                                        1\n",
      "Uncertain_significance,_risk_factor                                                             1\n",
      "Pathogenic,_association,_protective                                                             1\n",
      "Conflicting_interpretations_of_pathogenicity,_Affects,_other                                    1\n",
      "Conflicting_interpretations_of_pathogenicity,_protective                                        1\n",
      "Benign,_other                                                                                   1\n",
      "Likely_benign,_other                                                                            1\n",
      "Affects,_risk_factor                                                                            1\n",
      "Benign/Likely_benign,_protective                                                                1\n",
      "Conflicting_interpretations_of_pathogenicity,_Affects                                           1\n",
      "Conflicting_interpretations_of_pathogenicity,_association                                       1\n",
      "Benign/Likely_benign,_association                                                               1\n",
      "Conflicting_interpretations_of_pathogenicity,_other,_risk_factor                                1\n",
      "Conflicting_interpretations_of_pathogenicity,_Affects,_association,_drug_response,_other        1\n",
      "Likely_pathogenic,_association                                                                  1\n",
      "Benign,_drug_response                                                                           1\n",
      "Pathogenic,_protective                                                                          1\n",
      "Name: clinical_significant, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "### Data Exploratory\n",
    "# Print out all clinical significant values and counts for each value\n",
    "print('Clinical significant values:\\n')\n",
    "print(df.clinical_significant.describe())\n",
    "print('')\n",
    "print(df.clinical_significant.value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Disease name values:\n",
      "\n",
      "count            102321\n",
      "unique             6140\n",
      "top       not_specified\n",
      "freq              14400\n",
      "Name: disease_name, dtype: object\n",
      "\n",
      "not_specified                                                        14400\n",
      "Not_Given                                                            12670\n",
      "Hereditary_cancer-predisposing_syndrome                               2951\n",
      "Limb-girdle_muscular_dystrophy,_type_2J|Dilated_cardiomyopathy_1G     2588\n",
      "Lynch_syndrome                                                        1717\n",
      "not_specified|not_provided                                            1190\n",
      "Hereditary_nonpolyposis_colon_cancer                                  1095\n",
      "Alstrom_syndrome                                                       846\n",
      "Nemaline_myopathy_2                                                    769\n",
      "Cardiovascular_phenotype                                               742\n",
      "Name: disease_name, dtype: int64\n",
      "\n",
      "Gene Name:\n",
      "count                         102321\n",
      "unique                          1512\n",
      "top       TTN:7273|TTN-AS1:100506866\n",
      "freq                            6294\n",
      "Name: gene_name, dtype: object\n",
      "\n",
      "Mutation Value Reference:\n",
      "count     102321\n",
      "unique      2032\n",
      "top            C\n",
      "freq       31467\n",
      "Name: mutation_value_ref, dtype: object\n",
      "\n",
      "Mutation Value Alternate:\n",
      "count     102321\n",
      "unique      1196\n",
      "top            T\n",
      "freq       30115\n",
      "Name: mutation_value_alt, dtype: object\n"
     ]
    }
   ],
   "source": [
    "## Data Exploratory\n",
    "\n",
    "# Clinical significant\n",
    "print('Disease name values:\\n')\n",
    "print(df.disease_name.describe())\n",
    "print('')\n",
    "print(df.disease_name.value_counts().head(10))\n",
    "\n",
    "print('\\nGene Name:')\n",
    "print(df.gene_name.describe())\n",
    "\n",
    "print('\\nMutation Value Reference:')\n",
    "print(df.mutation_value_ref.describe())\n",
    "\n",
    "print('\\nMutation Value Alternate:')\n",
    "print(df.mutation_value_alt.describe())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Extract dangerous gene & mutation combination:\n",
    "# Consider it's dangerous when containing \"pathogenic\" in the clinical significant \n",
    "df2 = df[df.clinical_significant.str.lower().str.contains('pathogenic')]\n",
    "\n",
    "# And it does not contain \"conflicting_interpretations_of_pathogenicity\" \n",
    "df2 = df2[~df2.clinical_significant.str.lower().str.contains('pathogenicity')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Disease Name checking:\n",
      "count         19474\n",
      "unique         2976\n",
      "top       Not_Given\n",
      "freq           3712\n",
      "Name: disease_name, dtype: object\n",
      "Not_Given                                                 3712\n",
      "Lynch_syndrome                                             788\n",
      "Hereditary_cancer-predisposing_syndrome                    432\n",
      "Ehlers-Danlos_syndrome,_type_4                             418\n",
      "Primary_pulmonary_hypertension                             305\n",
      "Nemaline_myopathy_2                                        234\n",
      "Inborn_genetic_diseases                                    201\n",
      "Severe_myoclonic_epilepsy_in_infancy                       190\n",
      "Usher_syndrome,_type_2A|Retinitis_pigmentosa_39            181\n",
      "Hereditary_nonpolyposis_colon_cancer                       178\n",
      "Primary_hyperoxaluria,_type_I                              175\n",
      "Primary_dilated_cardiomyopathy                             165\n",
      "Alstrom_syndrome                                           156\n",
      "Biotinidase_deficiency                                     140\n",
      "Hereditary_cancer-predisposing_syndrome|Lynch_syndrome     128\n",
      "Name: disease_name, dtype: int64\n",
      "\n",
      "Gene_name without Not_Given entries:\n",
      "2\n",
      "\n",
      "Mutation_ID without Not_Given entries:\n",
      "0\n",
      "\n",
      "Chromosome without Not_Given entries:\n",
      "0\n",
      "\n",
      "Mutation position without Not_Given entries:\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "## Sanity Check after filtering:\n",
    "print('Disease Name checking:')\n",
    "print(df2.disease_name.describe())\n",
    "print(df2.disease_name.value_counts().head(15))\n",
    "\n",
    "print('\\nGene_name without Not_Given entries:')\n",
    "print(df2.gene_name[df2.gene_name == 'Not_Given'].count())\n",
    "\n",
    "print('\\nMutation_ID without Not_Given entries:')\n",
    "print(df2.mutation_ID[df2.mutation_ID == 'Not_Given'].count())\n",
    "\n",
    "print('\\nChromosome without Not_Given entries:')\n",
    "print(df2.chromosome[df2.chromosome == 'Not_Given'].count())\n",
    "\n",
    "print('\\nMutation position without Not_Given entries:')\n",
    "print(df2.mutation_position[df2.mutation_position == 'Not_Given'].count())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Clinical Significant checking:\n",
      "count          15750\n",
      "unique            13\n",
      "top       Pathogenic\n",
      "freq           10418\n",
      "Name: clinical_significant, dtype: object\n",
      "Pathogenic                                   10418\n",
      "Likely_pathogenic                             4458\n",
      "Pathogenic/Likely_pathogenic                   836\n",
      "Pathogenic,_risk_factor                         11\n",
      "Pathogenic/Likely_pathogenic,_other              8\n",
      "Pathogenic/Likely_pathogenic,_risk_factor        6\n",
      "Likely_pathogenic,_risk_factor                   5\n",
      "Pathogenic,_Affects                              2\n",
      "Pathogenic,_other                                2\n",
      "Pathogenic,_association,_protective              1\n",
      "Likely_pathogenic,_association                   1\n",
      "Likely_pathogenic,_other                         1\n",
      "Pathogenic,_protective                           1\n",
      "Name: clinical_significant, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mutation_ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>mutation_position</th>\n",
       "      <th>mutation_value_ref</th>\n",
       "      <th>mutation_value_alt</th>\n",
       "      <th>clinical_significant</th>\n",
       "      <th>disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>183381</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>161455</td>\n",
       "      <td>1</td>\n",
       "      <td>1014316</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>161454</td>\n",
       "      <td>1</td>\n",
       "      <td>1014359</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>243036</td>\n",
       "      <td>1</td>\n",
       "      <td>1022225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>243037</td>\n",
       "      <td>1</td>\n",
       "      <td>1022313</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15745</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>39706</td>\n",
       "      <td>3</td>\n",
       "      <td>179210289</td>\n",
       "      <td>TAGA</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Megalencephaly_cutis_marmorata_telangiectatica...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15746</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>376470</td>\n",
       "      <td>3</td>\n",
       "      <td>179210291</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Hepatocellular_carcinoma|Transitional_cell_car...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15747</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>376471</td>\n",
       "      <td>3</td>\n",
       "      <td>179210291</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Hepatocellular_carcinoma|Transitional_cell_car...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15748</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>45465</td>\n",
       "      <td>3</td>\n",
       "      <td>179210292</td>\n",
       "      <td>AAGATTTGCTGAACCC</td>\n",
       "      <td>A</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Non-small_cell_lung_cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15749</th>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>376472</td>\n",
       "      <td>3</td>\n",
       "      <td>179210293</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Hepatocellular_carcinoma|Transitional_cell_car...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15750 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene_name  mutation_ID  chromosome  mutation_position  \\\n",
       "0       ISG15:9636       183381           1            1014143   \n",
       "1       ISG15:9636       161455           1            1014316   \n",
       "2       ISG15:9636       161454           1            1014359   \n",
       "3      AGRN:375790       243036           1            1022225   \n",
       "4      AGRN:375790       243037           1            1022313   \n",
       "...            ...          ...         ...                ...   \n",
       "15745  PIK3CA:5290        39706           3          179210289   \n",
       "15746  PIK3CA:5290       376470           3          179210291   \n",
       "15747  PIK3CA:5290       376471           3          179210291   \n",
       "15748  PIK3CA:5290        45465           3          179210292   \n",
       "15749  PIK3CA:5290       376472           3          179210293   \n",
       "\n",
       "      mutation_value_ref mutation_value_alt clinical_significant  \\\n",
       "0                      C                  T           Pathogenic   \n",
       "1                      C                 CG           Pathogenic   \n",
       "2                      G                  T           Pathogenic   \n",
       "3                      G                  A           Pathogenic   \n",
       "4                      A                  T           Pathogenic   \n",
       "...                  ...                ...                  ...   \n",
       "15745               TAGA                  T           Pathogenic   \n",
       "15746                  G                  A    Likely_pathogenic   \n",
       "15747                  G                  C    Likely_pathogenic   \n",
       "15748   AAGATTTGCTGAACCC                  A    Likely_pathogenic   \n",
       "15749                  A                  T    Likely_pathogenic   \n",
       "\n",
       "                                            disease_name  \n",
       "0      Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "1      Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "2      Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "3                         Congenital_myasthenic_syndrome  \n",
       "4                         Congenital_myasthenic_syndrome  \n",
       "...                                                  ...  \n",
       "15745  Megalencephaly_cutis_marmorata_telangiectatica...  \n",
       "15746  Hepatocellular_carcinoma|Transitional_cell_car...  \n",
       "15747  Hepatocellular_carcinoma|Transitional_cell_car...  \n",
       "15748                         Non-small_cell_lung_cancer  \n",
       "15749  Hepatocellular_carcinoma|Transitional_cell_car...  \n",
       "\n",
       "[15750 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Remove \"Not_Given\" or \"not_specified\" from dataframe due to having no associated information\n",
    "df2 = df2[~df2.disease_name.isin(['Not_Given', 'not_specified','.'])]\n",
    "df2 = df2[~df2.gene_name.isin(['Not_Given','not_specified','.'])]\n",
    "df2 = df2.reset_index()\n",
    "df2 = df2.drop('index', axis=1)\n",
    "print('\\nClinical Significant checking:')\n",
    "print(df2.clinical_significant.describe())\n",
    "print(df2.clinical_significant.value_counts())\n",
    "df2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### My Assumptions:\n",
    "\n",
    "The \"INFOR\" data does not does not follow a particular order, but the variables are separated by a \";\" and a variable name and its values are separated by a \"=\" in between of them: variable_name=variable_value. We could extract the value for each variable using regular expression extract method.\n",
    "\n",
    "Variables of interest:\n",
    "- Gene name: the \"GENEINFO\" data in the \"INFOR\" column contains the gene symbol and the gene id. The gene symbol and id are separated by a colon (:), However, because there could be mutiple genes in an entry, therefore, I extract the complete data from the \"GENEINFO\" and do not separate between gene symbol & id. \n",
    "- Mutation ID number: is the \"ID\" column\n",
    "- Mutation position: includes the chromosome \"CHROM\" column and position \"POS\" column\n",
    "- Mutation value: includes reference base \"REF\" column and alternate base \"ALT\" column\n",
    "- Clinical significant: is the \"CLNSIG\" variable in the \"INFOR\" column\n",
    "- Disease that is implicated: is the \"CLNDN\" variable in the \"INFOR\" column\n",
    "\n",
    "Other assumptions:\n",
    "- Mutation position (POS): assume the position should be integer only, and some wrong entries with \"O\" should be replaced by '0' (zero).\n",
    "- The mutation values for reference and alternative bases (REF & ALT) as listed in the dataset are correct, a letter could be used multiple times in the value and it does not need to be corrected.\n",
    "\n",
    "Dangerous mutations:\n",
    "- Assume the dangerous gene and mutation combination contains the following words in its clinical significant description: \"pathogenic\", \"likely_pathogenic\". \n",
    "- Any clinical significant having \"conflicting_interpretations_of_pathogenicity\", \"risk_factor\", \"association\", \"affects\" and not having the sigle word \"pathogenic\" will not be considered as dangerous mutation, due to being on debate and not confirmed as the causes of the associated diseases.\n",
    "- Any record has \"pathogenic\" in the clinical significant but does not have data for the disease name, or gene name (or listed as \"Not_Given\") is excluded from the list of dangerous gene and mutation combination. Due to unclear disease or lacking of gene name information."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Final Result\n",
    "\n",
    "Below is the table for the first 100 harmful mutations to be present to my manager"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Total of harmful mutations: 15750\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mutation_ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>mutation_position</th>\n",
       "      <th>mutation_value_ref</th>\n",
       "      <th>mutation_value_alt</th>\n",
       "      <th>clinical_significant</th>\n",
       "      <th>disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>183381</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>161455</td>\n",
       "      <td>1</td>\n",
       "      <td>1014316</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>161454</td>\n",
       "      <td>1</td>\n",
       "      <td>1014359</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>243036</td>\n",
       "      <td>1</td>\n",
       "      <td>1022225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>243037</td>\n",
       "      <td>1</td>\n",
       "      <td>1022313</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>574478</td>\n",
       "      <td>1</td>\n",
       "      <td>1041354</td>\n",
       "      <td>CGCCCGCCAGGAGAATGTCTTCAAGAAGTTCGACG</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Myasthenic_syndrome,_congenital,_8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>126556</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome|Myasthenic_synd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>243038</td>\n",
       "      <td>1</td>\n",
       "      <td>1042136</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>243039</td>\n",
       "      <td>1</td>\n",
       "      <td>1050473</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>18241</td>\n",
       "      <td>1</td>\n",
       "      <td>1050575</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome|Myasthenic_synd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>126555</td>\n",
       "      <td>1</td>\n",
       "      <td>1050763</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome|Myasthenic_synd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>243040</td>\n",
       "      <td>1</td>\n",
       "      <td>1051775</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TNFRSF4:7293</td>\n",
       "      <td>96692</td>\n",
       "      <td>1</td>\n",
       "      <td>1213738</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Immunodeficiency_16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60484</td>\n",
       "      <td>1</td>\n",
       "      <td>1232279</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60493</td>\n",
       "      <td>1</td>\n",
       "      <td>1232294</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60488</td>\n",
       "      <td>1</td>\n",
       "      <td>1232471</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60489</td>\n",
       "      <td>1</td>\n",
       "      <td>1232478</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60497</td>\n",
       "      <td>1</td>\n",
       "      <td>1232594</td>\n",
       "      <td>CGGCGCGCCCTGGAGCGGGAGCA</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60490</td>\n",
       "      <td>1</td>\n",
       "      <td>1232630</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60494</td>\n",
       "      <td>1</td>\n",
       "      <td>1232684</td>\n",
       "      <td>GTGCTGGCCA</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60486</td>\n",
       "      <td>1</td>\n",
       "      <td>1232744</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>522415</td>\n",
       "      <td>1</td>\n",
       "      <td>1232834</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60492</td>\n",
       "      <td>1</td>\n",
       "      <td>1232859</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60495</td>\n",
       "      <td>1</td>\n",
       "      <td>1232897</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60496</td>\n",
       "      <td>1</td>\n",
       "      <td>1232927</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60485</td>\n",
       "      <td>1</td>\n",
       "      <td>1232972</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60487</td>\n",
       "      <td>1</td>\n",
       "      <td>1233177</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>60491</td>\n",
       "      <td>1</td>\n",
       "      <td>1233203</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>488046</td>\n",
       "      <td>1</td>\n",
       "      <td>1337992</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>488048</td>\n",
       "      <td>1</td>\n",
       "      <td>1337992</td>\n",
       "      <td>TCCCGGTGCTGCCGCTG</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>488045</td>\n",
       "      <td>1</td>\n",
       "      <td>1337999</td>\n",
       "      <td>G</td>\n",
       "      <td>GCTGCC</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>208047</td>\n",
       "      <td>1</td>\n",
       "      <td>1338000</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>208050</td>\n",
       "      <td>1</td>\n",
       "      <td>1338033</td>\n",
       "      <td>TAGGCAGG</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>208043</td>\n",
       "      <td>1</td>\n",
       "      <td>1338045</td>\n",
       "      <td>AA</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>208049</td>\n",
       "      <td>1</td>\n",
       "      <td>1338053</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>208048</td>\n",
       "      <td>1</td>\n",
       "      <td>1338086</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>219223</td>\n",
       "      <td>1</td>\n",
       "      <td>1338092</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>208045</td>\n",
       "      <td>1</td>\n",
       "      <td>1338096</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>208044</td>\n",
       "      <td>1</td>\n",
       "      <td>1338098</td>\n",
       "      <td>GGGGGCAGCCGGGT</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>208046</td>\n",
       "      <td>1</td>\n",
       "      <td>1338107</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>488047</td>\n",
       "      <td>1</td>\n",
       "      <td>1338107</td>\n",
       "      <td>CGGGTGGGGCAGCG</td>\n",
       "      <td>C</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>520999</td>\n",
       "      <td>1</td>\n",
       "      <td>1338111</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>402162</td>\n",
       "      <td>1</td>\n",
       "      <td>1353913</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Abnormality_of_brain_morphology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>ATAD3A:55210</td>\n",
       "      <td>592109</td>\n",
       "      <td>1</td>\n",
       "      <td>1517250</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Harel-Yoon_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>ATAD3A:55210</td>\n",
       "      <td>225696</td>\n",
       "      <td>1</td>\n",
       "      <td>1529299</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Harel-Yoon_syndrome|not_provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>ATAD3A:55210</td>\n",
       "      <td>522903</td>\n",
       "      <td>1</td>\n",
       "      <td>1534004</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Harel-Yoon_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>161196</td>\n",
       "      <td>1</td>\n",
       "      <td>1535370</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>161192</td>\n",
       "      <td>1</td>\n",
       "      <td>1535372</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Spinocerebellar_ataxia_21|not_provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>161193</td>\n",
       "      <td>1</td>\n",
       "      <td>1535392</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>161194</td>\n",
       "      <td>1</td>\n",
       "      <td>1535616</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>523023</td>\n",
       "      <td>1</td>\n",
       "      <td>1535619</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>161195</td>\n",
       "      <td>1</td>\n",
       "      <td>1535723</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>521025</td>\n",
       "      <td>1</td>\n",
       "      <td>1787340</td>\n",
       "      <td>GATC</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>224718</td>\n",
       "      <td>1</td>\n",
       "      <td>1787378</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Intell...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>521213</td>\n",
       "      <td>1</td>\n",
       "      <td>1804496</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>224717</td>\n",
       "      <td>1</td>\n",
       "      <td>1804548</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Intell...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>224716</td>\n",
       "      <td>1</td>\n",
       "      <td>1804565</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Strabi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>208722</td>\n",
       "      <td>1</td>\n",
       "      <td>1806503</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Cleft_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>224715</td>\n",
       "      <td>1</td>\n",
       "      <td>1806503</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Seizur...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>224714</td>\n",
       "      <td>1</td>\n",
       "      <td>1806509</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Strabi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>431082</td>\n",
       "      <td>1</td>\n",
       "      <td>1806512</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Mental_retardation,_autosomal_dominant_42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>224713</td>\n",
       "      <td>1</td>\n",
       "      <td>1806513</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Seizur...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>224712</td>\n",
       "      <td>1</td>\n",
       "      <td>1806514</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Strabi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>224711</td>\n",
       "      <td>1</td>\n",
       "      <td>1806515</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Intell...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>617622</td>\n",
       "      <td>1</td>\n",
       "      <td>1815801</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Mental_retardation,_autosomal_dominant_42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>224869</td>\n",
       "      <td>1</td>\n",
       "      <td>2228828</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>570752</td>\n",
       "      <td>1</td>\n",
       "      <td>2228834</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>37259</td>\n",
       "      <td>1</td>\n",
       "      <td>2228860</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>37261</td>\n",
       "      <td>1</td>\n",
       "      <td>2228866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Inborn_genetic_diseases|Shprintzen-Goldberg_sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>37262</td>\n",
       "      <td>1</td>\n",
       "      <td>2228866</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Inborn_genetic_diseases|Shprintzen-Goldberg_sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>37260</td>\n",
       "      <td>1</td>\n",
       "      <td>2228867</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>39783</td>\n",
       "      <td>1</td>\n",
       "      <td>2228867</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>39785</td>\n",
       "      <td>1</td>\n",
       "      <td>2228870</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>37263</td>\n",
       "      <td>1</td>\n",
       "      <td>2229045</td>\n",
       "      <td>GTCCGACCGC</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>39784</td>\n",
       "      <td>1</td>\n",
       "      <td>2229045</td>\n",
       "      <td>GTCCGACCGCTCC</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>37258</td>\n",
       "      <td>1</td>\n",
       "      <td>2229113</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>224868</td>\n",
       "      <td>1</td>\n",
       "      <td>2229115</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome|not_provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>222819</td>\n",
       "      <td>1</td>\n",
       "      <td>2229118</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>495053</td>\n",
       "      <td>1</td>\n",
       "      <td>2229305</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Disproportionate_tall_stature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>463409</td>\n",
       "      <td>1</td>\n",
       "      <td>2229401</td>\n",
       "      <td>AGCTCAGCGAGCGCAGCGTCCGCGTGTACCACGAGTGCTTCGGCAA...</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>554342</td>\n",
       "      <td>1</td>\n",
       "      <td>2406483</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>557020</td>\n",
       "      <td>1</td>\n",
       "      <td>2406483</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>6771</td>\n",
       "      <td>1</td>\n",
       "      <td>2406526</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>555297</td>\n",
       "      <td>1</td>\n",
       "      <td>2406580</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>296273</td>\n",
       "      <td>1</td>\n",
       "      <td>2406580</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Zellweger_syndrome|Peroxisome_biogenesis_disor...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>553074</td>\n",
       "      <td>1</td>\n",
       "      <td>2406599</td>\n",
       "      <td>GCT</td>\n",
       "      <td>G</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>553702</td>\n",
       "      <td>1</td>\n",
       "      <td>2406606</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>555007</td>\n",
       "      <td>1</td>\n",
       "      <td>2406718</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>553321</td>\n",
       "      <td>1</td>\n",
       "      <td>2406718</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>225040</td>\n",
       "      <td>1</td>\n",
       "      <td>2406719</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>554630</td>\n",
       "      <td>1</td>\n",
       "      <td>2406734</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>550383</td>\n",
       "      <td>1</td>\n",
       "      <td>2406739</td>\n",
       "      <td>GGT</td>\n",
       "      <td>G</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>162435</td>\n",
       "      <td>1</td>\n",
       "      <td>2406766</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>6774</td>\n",
       "      <td>1</td>\n",
       "      <td>2406791</td>\n",
       "      <td>C</td>\n",
       "      <td>CT</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder,_complementatio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>557464</td>\n",
       "      <td>1</td>\n",
       "      <td>2406792</td>\n",
       "      <td>TGCAGCCCCATGG</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>551969</td>\n",
       "      <td>1</td>\n",
       "      <td>2406917</td>\n",
       "      <td>ACT</td>\n",
       "      <td>A</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>550051</td>\n",
       "      <td>1</td>\n",
       "      <td>2406920</td>\n",
       "      <td>CA</td>\n",
       "      <td>GAG</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>556606</td>\n",
       "      <td>1</td>\n",
       "      <td>2406931</td>\n",
       "      <td>TGA</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>553280</td>\n",
       "      <td>1</td>\n",
       "      <td>2408450</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>6770</td>\n",
       "      <td>1</td>\n",
       "      <td>2408451</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene_name  mutation_ID  chromosome  mutation_position  \\\n",
       "0       ISG15:9636       183381           1            1014143   \n",
       "1       ISG15:9636       161455           1            1014316   \n",
       "2       ISG15:9636       161454           1            1014359   \n",
       "3      AGRN:375790       243036           1            1022225   \n",
       "4      AGRN:375790       243037           1            1022313   \n",
       "5      AGRN:375790       574478           1            1041354   \n",
       "6      AGRN:375790       126556           1            1041582   \n",
       "7      AGRN:375790       243038           1            1042136   \n",
       "8      AGRN:375790       243039           1            1050473   \n",
       "9      AGRN:375790        18241           1            1050575   \n",
       "10     AGRN:375790       126555           1            1050763   \n",
       "11     AGRN:375790       243040           1            1051775   \n",
       "12    TNFRSF4:7293        96692           1            1213738   \n",
       "13  B3GALT6:126792        60484           1            1232279   \n",
       "14  B3GALT6:126792        60493           1            1232294   \n",
       "15  B3GALT6:126792        60488           1            1232471   \n",
       "16  B3GALT6:126792        60489           1            1232478   \n",
       "17  B3GALT6:126792        60497           1            1232594   \n",
       "18  B3GALT6:126792        60490           1            1232630   \n",
       "19  B3GALT6:126792        60494           1            1232684   \n",
       "20  B3GALT6:126792        60486           1            1232744   \n",
       "21  B3GALT6:126792       522415           1            1232834   \n",
       "22  B3GALT6:126792        60492           1            1232859   \n",
       "23  B3GALT6:126792        60495           1            1232897   \n",
       "24  B3GALT6:126792        60496           1            1232927   \n",
       "25  B3GALT6:126792        60485           1            1232972   \n",
       "26  B3GALT6:126792        60487           1            1233177   \n",
       "27  B3GALT6:126792        60491           1            1233203   \n",
       "28       DVL1:1855       488046           1            1337992   \n",
       "29       DVL1:1855       488048           1            1337992   \n",
       "30       DVL1:1855       488045           1            1337999   \n",
       "31       DVL1:1855       208047           1            1338000   \n",
       "32       DVL1:1855       208050           1            1338033   \n",
       "33       DVL1:1855       208043           1            1338045   \n",
       "34       DVL1:1855       208049           1            1338053   \n",
       "35       DVL1:1855       208048           1            1338086   \n",
       "36       DVL1:1855       219223           1            1338092   \n",
       "37       DVL1:1855       208045           1            1338096   \n",
       "38       DVL1:1855       208044           1            1338098   \n",
       "39       DVL1:1855       208046           1            1338107   \n",
       "40       DVL1:1855       488047           1            1338107   \n",
       "41       DVL1:1855       520999           1            1338111   \n",
       "42     MXRA8:54587       402162           1            1353913   \n",
       "43    ATAD3A:55210       592109           1            1517250   \n",
       "44    ATAD3A:55210       225696           1            1529299   \n",
       "45    ATAD3A:55210       522903           1            1534004   \n",
       "46  TMEM240:339453       161196           1            1535370   \n",
       "47  TMEM240:339453       161192           1            1535372   \n",
       "48  TMEM240:339453       161193           1            1535392   \n",
       "49  TMEM240:339453       161194           1            1535616   \n",
       "50  TMEM240:339453       523023           1            1535619   \n",
       "51  TMEM240:339453       161195           1            1535723   \n",
       "52       GNB1:2782       521025           1            1787340   \n",
       "53       GNB1:2782       224718           1            1787378   \n",
       "54       GNB1:2782       521213           1            1804496   \n",
       "55       GNB1:2782       224717           1            1804548   \n",
       "56       GNB1:2782       224716           1            1804565   \n",
       "57       GNB1:2782       208722           1            1806503   \n",
       "58       GNB1:2782       224715           1            1806503   \n",
       "59       GNB1:2782       224714           1            1806509   \n",
       "60       GNB1:2782       431082           1            1806512   \n",
       "61       GNB1:2782       224713           1            1806513   \n",
       "62       GNB1:2782       224712           1            1806514   \n",
       "63       GNB1:2782       224711           1            1806515   \n",
       "64       GNB1:2782       617622           1            1815801   \n",
       "65        SKI:6497       224869           1            2228828   \n",
       "66        SKI:6497       570752           1            2228834   \n",
       "67        SKI:6497        37259           1            2228860   \n",
       "68        SKI:6497        37261           1            2228866   \n",
       "69        SKI:6497        37262           1            2228866   \n",
       "70        SKI:6497        37260           1            2228867   \n",
       "71        SKI:6497        39783           1            2228867   \n",
       "72        SKI:6497        39785           1            2228870   \n",
       "73        SKI:6497        37263           1            2229045   \n",
       "74        SKI:6497        39784           1            2229045   \n",
       "75        SKI:6497        37258           1            2229113   \n",
       "76        SKI:6497       224868           1            2229115   \n",
       "77        SKI:6497       222819           1            2229118   \n",
       "78        SKI:6497       495053           1            2229305   \n",
       "79        SKI:6497       463409           1            2229401   \n",
       "80      PEX10:5192       554342           1            2406483   \n",
       "81      PEX10:5192       557020           1            2406483   \n",
       "82      PEX10:5192         6771           1            2406526   \n",
       "83      PEX10:5192       555297           1            2406580   \n",
       "84      PEX10:5192       296273           1            2406580   \n",
       "85      PEX10:5192       553074           1            2406599   \n",
       "86      PEX10:5192       553702           1            2406606   \n",
       "87      PEX10:5192       555007           1            2406718   \n",
       "88      PEX10:5192       553321           1            2406718   \n",
       "89      PEX10:5192       225040           1            2406719   \n",
       "90      PEX10:5192       554630           1            2406734   \n",
       "91      PEX10:5192       550383           1            2406739   \n",
       "92      PEX10:5192       162435           1            2406766   \n",
       "93      PEX10:5192         6774           1            2406791   \n",
       "94      PEX10:5192       557464           1            2406792   \n",
       "95      PEX10:5192       551969           1            2406917   \n",
       "96      PEX10:5192       550051           1            2406920   \n",
       "97      PEX10:5192       556606           1            2406931   \n",
       "98      PEX10:5192       553280           1            2408450   \n",
       "99      PEX10:5192         6770           1            2408451   \n",
       "\n",
       "                                   mutation_value_ref mutation_value_alt  \\\n",
       "0                                                   C                  T   \n",
       "1                                                   C                 CG   \n",
       "2                                                   G                  T   \n",
       "3                                                   G                  A   \n",
       "4                                                   A                  T   \n",
       "5                 CGCCCGCCAGGAGAATGTCTTCAAGAAGTTCGACG                  C   \n",
       "6                                                   C                  T   \n",
       "7                                                   T                 TC   \n",
       "8                                                   G                  A   \n",
       "9                                                   G                  C   \n",
       "10                                                  G                  T   \n",
       "11                                                  G                  A   \n",
       "12                                                  G                  A   \n",
       "13                                                  A                  G   \n",
       "14                                                  C                  T   \n",
       "15                                                  A                  G   \n",
       "16                                                  C                  T   \n",
       "17                            CGGCGCGCCCTGGAGCGGGAGCA                  C   \n",
       "18                                                 GA                  G   \n",
       "19                                         GTGCTGGCCA                  G   \n",
       "20                                                  G                  A   \n",
       "21                                                  T                  C   \n",
       "22                                                 CG                  C   \n",
       "23                                                  G                  C   \n",
       "24                                                  G                  A   \n",
       "25                                                  C                  T   \n",
       "26                                                  G                  C   \n",
       "27                                                  T                  A   \n",
       "28                                                 TC                  T   \n",
       "29                                  TCCCGGTGCTGCCGCTG                  T   \n",
       "30                                                  G             GCTGCC   \n",
       "31                                                 CT                  C   \n",
       "32                                           TAGGCAGG                  C   \n",
       "33                                                 AA                  G   \n",
       "34                                                 TG                  T   \n",
       "35                                                 AC                  A   \n",
       "36                                                 AG                  A   \n",
       "37                                                 CA                  C   \n",
       "38                                     GGGGGCAGCCGGGT                  G   \n",
       "39                                                 CG                  C   \n",
       "40                                     CGGGTGGGGCAGCG                  C   \n",
       "41                                                 TG                  T   \n",
       "42                                                  A                  T   \n",
       "43                                                 GC                  G   \n",
       "44                                                  C                  T   \n",
       "45                                                  C                  T   \n",
       "46                                                  G                  A   \n",
       "47                                                  G                  A   \n",
       "48                                                  G                  C   \n",
       "49                                                  G                  A   \n",
       "50                                                  C                  T   \n",
       "51                                                  G                  A   \n",
       "52                                               GATC                  G   \n",
       "53                                                  C                  T   \n",
       "54                                                  T                  C   \n",
       "55                                                  T                  C   \n",
       "56                                                  A                  G   \n",
       "57                                                  A                  G   \n",
       "58                                                  A                  T   \n",
       "59                                                  T                  C   \n",
       "60                                                  C                  G   \n",
       "61                                                  C                  T   \n",
       "62                                                  A                  C   \n",
       "63                                                  T                  C   \n",
       "64                                                  C                  T   \n",
       "65                                                  T                  G   \n",
       "66                                                  A                  C   \n",
       "67                                                  C                  G   \n",
       "68                                                  G                  A   \n",
       "69                                                  G                  T   \n",
       "70                                                  G                  A   \n",
       "71                                                  G                  T   \n",
       "72                                                  C                  A   \n",
       "73                                         GTCCGACCGC                  G   \n",
       "74                                      GTCCGACCGCTCC                  G   \n",
       "75                                                  G                  A   \n",
       "76                                                  G                  C   \n",
       "77                                                  G                  A   \n",
       "78                                                  C                  A   \n",
       "79  AGCTCAGCGAGCGCAGCGTCCGCGTGTACCACGAGTGCTTCGGCAA...                  A   \n",
       "80                                                  C                  G   \n",
       "81                                                  C                  T   \n",
       "82                                                  G                  C   \n",
       "83                                                 CA                  C   \n",
       "84                                                CAG                  C   \n",
       "85                                                GCT                  G   \n",
       "86                                                  C                  A   \n",
       "87                                                  A                  G   \n",
       "88                                                  A                  T   \n",
       "89                                                  C                  G   \n",
       "90                                                 GC                  G   \n",
       "91                                                GGT                  G   \n",
       "92                                                  G                  A   \n",
       "93                                                  C                 CT   \n",
       "94                                      TGCAGCCCCATGG                  T   \n",
       "95                                                ACT                  A   \n",
       "96                                                 CA                GAG   \n",
       "97                                                TGA                  T   \n",
       "98                                                 AC                  A   \n",
       "99                                                  C                  T   \n",
       "\n",
       "            clinical_significant  \\\n",
       "0                     Pathogenic   \n",
       "1                     Pathogenic   \n",
       "2                     Pathogenic   \n",
       "3                     Pathogenic   \n",
       "4                     Pathogenic   \n",
       "5                     Pathogenic   \n",
       "6                     Pathogenic   \n",
       "7                     Pathogenic   \n",
       "8                     Pathogenic   \n",
       "9                     Pathogenic   \n",
       "10                    Pathogenic   \n",
       "11                    Pathogenic   \n",
       "12                    Pathogenic   \n",
       "13                    Pathogenic   \n",
       "14                    Pathogenic   \n",
       "15                    Pathogenic   \n",
       "16                    Pathogenic   \n",
       "17                    Pathogenic   \n",
       "18                    Pathogenic   \n",
       "19                    Pathogenic   \n",
       "20                    Pathogenic   \n",
       "21                    Pathogenic   \n",
       "22                    Pathogenic   \n",
       "23                    Pathogenic   \n",
       "24                    Pathogenic   \n",
       "25                    Pathogenic   \n",
       "26                    Pathogenic   \n",
       "27                    Pathogenic   \n",
       "28             Likely_pathogenic   \n",
       "29             Likely_pathogenic   \n",
       "30             Likely_pathogenic   \n",
       "31                    Pathogenic   \n",
       "32                    Pathogenic   \n",
       "33                    Pathogenic   \n",
       "34                    Pathogenic   \n",
       "35                    Pathogenic   \n",
       "36                    Pathogenic   \n",
       "37                    Pathogenic   \n",
       "38                    Pathogenic   \n",
       "39                    Pathogenic   \n",
       "40             Likely_pathogenic   \n",
       "41                    Pathogenic   \n",
       "42             Likely_pathogenic   \n",
       "43             Likely_pathogenic   \n",
       "44             Likely_pathogenic   \n",
       "45                    Pathogenic   \n",
       "46                    Pathogenic   \n",
       "47  Pathogenic/Likely_pathogenic   \n",
       "48                    Pathogenic   \n",
       "49                    Pathogenic   \n",
       "50                    Pathogenic   \n",
       "51                    Pathogenic   \n",
       "52                    Pathogenic   \n",
       "53  Pathogenic/Likely_pathogenic   \n",
       "54                    Pathogenic   \n",
       "55  Pathogenic/Likely_pathogenic   \n",
       "56  Pathogenic/Likely_pathogenic   \n",
       "57                    Pathogenic   \n",
       "58  Pathogenic/Likely_pathogenic   \n",
       "59                    Pathogenic   \n",
       "60             Likely_pathogenic   \n",
       "61             Likely_pathogenic   \n",
       "62  Pathogenic/Likely_pathogenic   \n",
       "63  Pathogenic/Likely_pathogenic   \n",
       "64             Likely_pathogenic   \n",
       "65                    Pathogenic   \n",
       "66             Likely_pathogenic   \n",
       "67                    Pathogenic   \n",
       "68  Pathogenic/Likely_pathogenic   \n",
       "69                    Pathogenic   \n",
       "70                    Pathogenic   \n",
       "71                    Pathogenic   \n",
       "72                    Pathogenic   \n",
       "73                    Pathogenic   \n",
       "74                    Pathogenic   \n",
       "75                    Pathogenic   \n",
       "76  Pathogenic/Likely_pathogenic   \n",
       "77             Likely_pathogenic   \n",
       "78             Likely_pathogenic   \n",
       "79                    Pathogenic   \n",
       "80             Likely_pathogenic   \n",
       "81             Likely_pathogenic   \n",
       "82                    Pathogenic   \n",
       "83             Likely_pathogenic   \n",
       "84                    Pathogenic   \n",
       "85             Likely_pathogenic   \n",
       "86             Likely_pathogenic   \n",
       "87             Likely_pathogenic   \n",
       "88             Likely_pathogenic   \n",
       "89                    Pathogenic   \n",
       "90  Pathogenic/Likely_pathogenic   \n",
       "91             Likely_pathogenic   \n",
       "92                    Pathogenic   \n",
       "93                    Pathogenic   \n",
       "94             Likely_pathogenic   \n",
       "95             Likely_pathogenic   \n",
       "96             Likely_pathogenic   \n",
       "97             Likely_pathogenic   \n",
       "98             Likely_pathogenic   \n",
       "99                    Pathogenic   \n",
       "\n",
       "                                         disease_name  \n",
       "0   Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "1   Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "2   Immunodeficiency_38_with_basal_ganglia_calcifi...  \n",
       "3                      Congenital_myasthenic_syndrome  \n",
       "4                      Congenital_myasthenic_syndrome  \n",
       "5                  Myasthenic_syndrome,_congenital,_8  \n",
       "6   Congenital_myasthenic_syndrome|Myasthenic_synd...  \n",
       "7                      Congenital_myasthenic_syndrome  \n",
       "8                      Congenital_myasthenic_syndrome  \n",
       "9   Congenital_myasthenic_syndrome|Myasthenic_synd...  \n",
       "10  Congenital_myasthenic_syndrome|Myasthenic_synd...  \n",
       "11                     Congenital_myasthenic_syndrome  \n",
       "12                                Immunodeficiency_16  \n",
       "13  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "14          Ehlers-Danlos_syndrome,_progeroid_type,_2  \n",
       "15  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "16  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "17  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "18          Ehlers-Danlos_syndrome,_progeroid_type,_2  \n",
       "19          Ehlers-Danlos_syndrome,_progeroid_type,_2  \n",
       "20  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "21  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "22          Ehlers-Danlos_syndrome,_progeroid_type,_2  \n",
       "23  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "24  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "25  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "26  Spondyloepimetaphyseal_dysplasia_with_joint_la...  \n",
       "27          Ehlers-Danlos_syndrome,_progeroid_type,_2  \n",
       "28             Robinow_syndrome,_autosomal_dominant_2  \n",
       "29             Robinow_syndrome,_autosomal_dominant_2  \n",
       "30             Robinow_syndrome,_autosomal_dominant_2  \n",
       "31             Robinow_syndrome,_autosomal_dominant_2  \n",
       "32             Robinow_syndrome,_autosomal_dominant_2  \n",
       "33             Robinow_syndrome,_autosomal_dominant_2  \n",
       "34             Robinow_syndrome,_autosomal_dominant_2  \n",
       "35             Robinow_syndrome,_autosomal_dominant_2  \n",
       "36             Robinow_syndrome,_autosomal_dominant_2  \n",
       "37             Robinow_syndrome,_autosomal_dominant_2  \n",
       "38             Robinow_syndrome,_autosomal_dominant_2  \n",
       "39             Robinow_syndrome,_autosomal_dominant_2  \n",
       "40             Robinow_syndrome,_autosomal_dominant_2  \n",
       "41                            Inborn_genetic_diseases  \n",
       "42                    Abnormality_of_brain_morphology  \n",
       "43                                Harel-Yoon_syndrome  \n",
       "44                   Harel-Yoon_syndrome|not_provided  \n",
       "45                                Harel-Yoon_syndrome  \n",
       "46                          Spinocerebellar_ataxia_21  \n",
       "47             Spinocerebellar_ataxia_21|not_provided  \n",
       "48                          Spinocerebellar_ataxia_21  \n",
       "49                          Spinocerebellar_ataxia_21  \n",
       "50                          Spinocerebellar_ataxia_21  \n",
       "51                          Spinocerebellar_ataxia_21  \n",
       "52                            Inborn_genetic_diseases  \n",
       "53  Neurodevelopmental_Disability|hypotonia|Intell...  \n",
       "54                            Inborn_genetic_diseases  \n",
       "55  Neurodevelopmental_Disability|hypotonia|Intell...  \n",
       "56  Neurodevelopmental_Disability|hypotonia|Strabi...  \n",
       "57  Neurodevelopmental_Disability|hypotonia|Cleft_...  \n",
       "58  Neurodevelopmental_Disability|hypotonia|Seizur...  \n",
       "59  Neurodevelopmental_Disability|hypotonia|Strabi...  \n",
       "60          Mental_retardation,_autosomal_dominant_42  \n",
       "61  Neurodevelopmental_Disability|hypotonia|Seizur...  \n",
       "62  Neurodevelopmental_Disability|hypotonia|Strabi...  \n",
       "63  Neurodevelopmental_Disability|hypotonia|Intell...  \n",
       "64          Mental_retardation,_autosomal_dominant_42  \n",
       "65                       Shprintzen-Goldberg_syndrome  \n",
       "66                       Shprintzen-Goldberg_syndrome  \n",
       "67                       Shprintzen-Goldberg_syndrome  \n",
       "68  Inborn_genetic_diseases|Shprintzen-Goldberg_sy...  \n",
       "69  Inborn_genetic_diseases|Shprintzen-Goldberg_sy...  \n",
       "70                       Shprintzen-Goldberg_syndrome  \n",
       "71                       Shprintzen-Goldberg_syndrome  \n",
       "72                       Shprintzen-Goldberg_syndrome  \n",
       "73                       Shprintzen-Goldberg_syndrome  \n",
       "74                       Shprintzen-Goldberg_syndrome  \n",
       "75                       Shprintzen-Goldberg_syndrome  \n",
       "76          Shprintzen-Goldberg_syndrome|not_provided  \n",
       "77                       Shprintzen-Goldberg_syndrome  \n",
       "78                      Disproportionate_tall_stature  \n",
       "79                       Shprintzen-Goldberg_syndrome  \n",
       "80  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "81  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "82                  Peroxisome_biogenesis_disorder_6B  \n",
       "83  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "84  Zellweger_syndrome|Peroxisome_biogenesis_disor...  \n",
       "85  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "86  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "87  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "88  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "89                            Inborn_genetic_diseases  \n",
       "90  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "91  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "92  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "93  Peroxisome_biogenesis_disorder,_complementatio...  \n",
       "94  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "95  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "96  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "97  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "98  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  \n",
       "99  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Print our the total number of harmful mutations and the first 100 harmful mutations\n",
    "print('\\nTotal of harmful mutations:', df2.shape[0])\n",
    "df2.head(100)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
